app/views/layouts/_page_header.html.slim

```diff
@@ -0,0 +1,9 @@
+- if @page_header
```

Comment:
My 2 cents about this approach: I would try to use `PageHeader` inside a React component instead rendering the page header in the layout using `react_component`. For example, An `<ApplicationPage>` component like 

```
 <div class='applications'>
 	<PageHeader ...>
 	<IndexTable ...>
 </div>
```

Then

```
react_component 'ApplicationPage', { results: @applications, fields: @fields, links: ['View Application'] }
```

we could create a <PageLayout> component that wrapps `PageHeader` if we want to remove duplication
```
<PageLayout ...>
 	<IndexTable ...>
</PageLayout>
```
Also, another component could wrap `PageHeaderSimple` or `PageHeader` and decide what to render.

I would do this because: 
* it looks like we are doing React all the way down, so we do not get married to `Rails` to handle our layouts.
* easier integration for `mobx` or any state managment 
* easier migration to SPA or migration to `Routes` using `react-routes`
* ...in a way, better jest test coverage

**However** I 100% feel this approach and I understand how it's removing duplication. It's not the end of world (or maybe it is :| ......... just kidding). I just think there is value in staying in React and not mixing Rails features and just try to use Rails as a backend API as much as possible.

As general rule I would to call `react_component` only once per request.

@chrisdolendo @creativecoder  any thoughts?

Reply:
+1. I'd prefer to keep React self contained for all the reasons stated.

Reply:
Sounds good! That's why I was trying so hard to get ahold of a React person to pair on this 😉 @fddayan, let's find time today or tmrw to pair on the React-y solution to this 🙂